prototype/mobile_endpoint/case/case_processing.py

```diff
@@ -0,0 +1,239 @@
+from collections import namedtuple
+import copy
+from datetime import datetime
+import logging
+
+from mobile_endpoint.case import const
+from mobile_endpoint.case.models import CommCareCase
+from mobile_endpoint.case.xml.parser import case_update_from_block
+from mobile_endpoint.dao import CaseDbCache
+from mobile_endpoint.exceptions import ReconciliationError
+from mobile_endpoint.form.models import XFormInstance
+
+
+logger = logging.getLogger(__name__)
+
+
+def process_cases_in_form(xform, dao):
+    with CaseDbCache(dao, domain=xform['domain'],
+                     lock=True, deleted_ok=True, xforms=[xform]) as case_db:
```

Comment:
out of curiosity is the `CaseDbCache` actually a useful abstraction on top of the `dao`? or was it just convenient since that was how it was already implemented